### PR TITLE
Docs: Convert to ReadTheDocs

### LINKED ISSUE #22: Add readthedocs and convert old docs into rst
Currently we have some small markdown docs which are not really extensive. A fully featured rst documentation would be better. Readthedocs fits this perfectly.

### PR DESCRIPTION
**Pull request by SchoolGuy**
This fixes #22

### CODE REVIEW

**Comment by SchoolGuy** [comment]
Don't know why rtd is refusing to build on my PR. Anyway the results can be seen here: https://orthos2-work-fork.readthedocs.io/en/docs-convert-to-rtd

**Comment by SchoolGuy** [comment]
So technically this is a 1:1 port to rtd from the markdown documents. But we need a lot of content. But this should not be the concern of this PR. Maybe this is the topic of a follow up PR.

**Comment by SchoolGuy** [comment]
I have received some internal documentation about Orthos2 at work. I will integrate that over the next few days.

**Comment by SchoolGuy** [comment]
The CI doesn't render this as I would like it to be rendered (and I think it should be rendered this way). I am investigating why this is the case.

File: docs/adminguide/setup.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+Installation/Setup
+******************
+
+Initial setup (development)
+###########################
+
+1. Prepare your local system:
+    .. code-block::
+
+        $ sudo zypper in python3-pip python3-virtualenv python3-devel libopenssl-devel
```
Comment (tomschr) [inline docs/adminguide/setup.rst]: You should also install the package `openldap2-devel` as the header file `lber.h` is needed:

```suggestion
        $ sudo zypper in python3-pip python3-virtualenv python3-devel libopenssl-devel openldap2-devel
```

File: docs/userguide.rst [hunk 2]
```diff
@@ -0,0 +1,615 @@
+**********
+User Guide
+**********
+
+Describes how to use Orthos as a user, how to find, resave and install machines for your purposes with the help of Web
```
Comment (tomschr) [inline docs/userguide.rst:5]: "Resave"? Never heard. :wink: 

```suggestion
Describes how to use Orthos as a user, how to find, save and install machines for your purposes with the help of Web
```

File: docs/adminguide.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
+.. _admin-guide:
+
+*********************
+Administrator`s Guide
+*********************
+
+These pages are partly very technical, you should read the user guide before and to have understood the basic concept of
+Orthos and what the goal of Orthos. Describes how to create and administrate groups and users, architecures, domains,
```
Comment (tomschr) [inline docs/adminguide.rst:8]:
```suggestion
These pages are partly very technical. You should read the User Guide before and to have understood the basic concept and the goals of Orthos. 

This document describes how to create and administrate groups and users, architectures, domains,
```

File: docs/userguide.rst [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
+
+
+Architecture
+############
+
+.. image:: img/userguide/00_architecture.png
+  :alt: Orthos2 High-Level Architecture
+
+Image Overall Architecture shows the overall architecture. All data is kept in the database. The Orthos Server is the
+central component of Orthos. It has following tasks:
```
Comment (tomschr) [inline docs/userguide.rst:38]:
```suggestion
central component of Orthos. It has the following tasks:
```

**Review by tomschr — COMMENTED** [review]
Just a few comments, I just had a brief look. 

Apart from that I have two additional comments:

1. Extend  `tox.ini` with `doc` target?

This could help to make things a bit easier as you only need to run `tox -e docs` and it builds your HTML file. Something like this:

```ini
[testenv:docs]
description = Build HTML documentation
basepython = python3
deps = -r{toxinidir}/docs/requirements.docs.txt
allowlist_externals =
    make
commands =
    make -C docs html
```

2. Investigate in diagram Sphinx extension?

There are some Sphinx extensions which could help to draw diagrams. This could make it easier to really have a diagram in these code blocks. But that is just a matter of taste.

File: docs/adminguide/setup.rst (hunk 1 above)
Comment (SchoolGuy) [reply docs/adminguide/setup.rst]: Done.

**Review by watologo1 — APPROVED** [review]
I incorporated Tom's fixes/suggestions and fixed up some typos myself.
I added 2 sentences about ldap authentication where appropriate, but did not change any conent/logic otherwise.

@aeisner  Thanks a lot Toni for the initial documention!

Thanks a lot @SchoolGuy for making it happen to show up in such a good shape!

It's very much appreciated!

**Pull request merged**